lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts

```diff
@@ -0,0 +1,138 @@
+import { SessionClaimValidator, SessionClaim } from "../session";
+import { MFAClaimValue, MFARequirementList, RecipeInterface } from "./types";
+import { checkFactorRequirement } from "./utils";
+
+/**
+ * We include "Class" in the class name, because it makes it easier to import/use the right thing (the instance exported by the recipe) instead of this.
+ * */
+export class MultiFactorAuthClaimClass implements SessionClaim<MFAClaimValue> {
+    public id = "st-mfa";
+    public validators: {
+        hasCompletedDefaultFactors: () => SessionClaimValidator;
+        hasCompletedFactors: (requirements: MFARequirementList) => SessionClaimValidator;
+    };
+
+    constructor(protected getRecipeImpl: () => RecipeInterface) {
+        this.validators = {
+            hasCompletedDefaultFactors: () => ({
+                id: this.id,
+                refresh: (...args) => this.refresh(...args),
+                shouldRefresh: (payload, userContext) => {
+                    const val = this.getValueFromPayload(payload, userContext);
+                    return !val;
+                },
+                validate: async (payload, userContext) => {
+                    const val = this.getValueFromPayload(payload, userContext);
+                    if (val === undefined) {
+                        return {
+                            isValid: false,
+                            reason: {
+                                message: "MFA info not available in payload",
+                            },
+                        };
+                    }
+
+                    if (val.v) {
+                        return {
+                            isValid: true,
+                        };
+                    }
+
+                    return {
+                        isValid: false,
+                        reason: {
+                            message: "not all required factors have been completed",
+                        },
+                    };
+                },
+            }),
+
+            hasCompletedFactors: (requirements: MFARequirementList) => ({
+                id: this.id,
+                shouldRefresh: (payload, userContext) => {
+                    const val = this.getValueFromPayload(payload, userContext);
+                    return !val;
+                },
+                refresh: (...args) => this.refresh(...args),
+                validate: (payload, userContext) => {
+                    const val = this.getValueFromPayload(payload, userContext);
+                    if (!val) {
+                        return {
+                            isValid: false,
+                            reason: {
+                                message: "MFA info not available in payload",
+                            },
+                        };
```

Comment:
Can this only come here if the backend is an older one that doesn't have MFA? in that case, do you want to actually return true cause the backend itself doesn't support MFA? If not, then whats the effect of returning false?

Reply:
@porcellus ?

Reply:
So this will only happen when the backend is an older version or does not have mfa, and the frontend will then show an access denied screen.